src/QMCHamiltonians/ECPotentialBuilder.cpp

```diff
@@ -166,10 +168,16 @@ bool ECPotentialBuilder::put(xmlNodePtr cur)
     }
     app_log() << "\n  Using SOECP potential \n"
               << "    Maximum grid on a sphere for SOECPotential: " << nknot_max << std::endl;
-    app_log() << "    Maximum grid for Simpson's rule for spin integral: " << sknot_max << std::endl;
+    if (spin_integrator == "exact")
+    {
+      app_log() << "    Using fast SOECP evaluation. Spin integration is exact" << std::endl;
+      apot->setFastEvaluation(true);
```

Comment:
Do you need fast and slow code paths both active in a single object?
If not, Can this option be passed via constructor?

Reply:
will fix this next, want to make sure tests pass first.

Reply:
I fixed the complex build. When you remove this setting, please make sure
https://github.com/QMCPACK/qmcpack/blob/592182424c27eda9caf049865da84aca11505607/src/QMCHamiltonians/SOECPComponent.cpp#L67
being reactive to the fast algorithm selection and set `total_knots_` properly.

Reply:
This looks to be taken care of now from what I can tell. Should fix the failing test as well